### PR TITLE
[BFCL] Add gemini-1.5-pro-002, gemini-1.5-pro-002-FC, gemini-1.5-pro-001, gemini-1.5-pro-001-FC, gemini-1.5-flash-002, gemini-1.5-flash-002-FC, gemini-1.0-pro-002, gemini-1.0-pro-002-FC

### PR DESCRIPTION
**Pull request by HuanzhiMao**
This PR adds the following new models to the leaderboard:

  - `gemini-1.5-pro-002`
  - `gemini-1.5-pro-002-FC`
  - `gemini-1.5-pro-001`
  - `gemini-1.5-pro-001-FC`
  - `gemini-1.5-flash-002`
  - `gemini-1.5-flash-002-FC`
  - `gemini-1.0-pro-002`
  - `gemini-1.0-pro-002-FC`

Note: All the code for Gemini prompting series inference are already written out in #644, but they were not thoroughly verified at that time due to limited bandwidth. This PR has the code tested and thus we 'officially' add support for the prompting version of the Gemini models.

### CODE REVIEW

**Comment by HuanzhiMao** [comment]
Update: Add `gemini-1.5-pro-002` and `gemini-1.5-pro-002_FC` in this PR as well.

**Comment by HuanzhiMao** [comment]
Hey @vandyxiaowei, would you mind taking a look at this PR? The Vertex AI library is a bit tricky to deal with, and I want to make sure I didn't mess up stuff. Thanks!

File: berkeley-function-call-leaderboard/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ All notable changes to the Berkeley Function Calling Leaderboard will be documen
   - `microsoft/Phi-3-mini-128k-instruct`
   - `microsoft/Phi-3-mini-4k-instruct`
 - [Sept 25, 2024] [#660](https://github.com/ShishirPatil/gorilla/pull/660): Bug fix in `parse_nested_value` function to handle nested dictionary values properly. 
+- [Sept 24, 2024] [#648](https://github.com/ShishirPatil/gorilla/pull/648): Add the following new models to the leaderboard:
+  - `gemini-1.5-pro-002`
+  - `gemini-1.5-pro-002-FC`
+  - `gemini-1.5-pro-preview-0514`
```
Comment (naseer9shaik) [inline berkeley-function-call-leaderboard/CHANGELOG.md:17]: Google now launched GA versions we don't need to include model with "preview" tag in the model name. So I think it is enough to include the below models:
1. gemini-1.5-flash-001 (stable version launched in May ) and gemini-1.5-flash-002 (stable version last week)
2.  gemini-1.5-pro-001 (stable version model launched in May ) and gemini-1.5-pro-002 (stable version last week)
3.  gemini-1.0-pro-001 (stable version model launched in Feb ) and gemini-1.0-pro-002 (stable version in Apr) 

All of the above models support function calling. Could we please add the above models to leaderboard (if not already included):

Sources: https://cloud.google.com/vertex-ai/generative-ai/docs/learn/model-versions#stable-versions-available and https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/function-calling#supported_models

Comment (HuanzhiMao) [reply berkeley-function-call-leaderboard/CHANGELOG.md:17]: Sure. We can use the GA versions instead. 
(I was previously under the impression that the `preview` ones are better than the GA ones.)

Comment (naseer9shaik) [reply berkeley-function-call-leaderboard/CHANGELOG.md:17]: This PR looks good to me, will you update this PR with the GA models?

Comment (HuanzhiMao) [reply berkeley-function-call-leaderboard/CHANGELOG.md]: Yes. I have updated the PR with the GA models.

File: berkeley-function-call-leaderboard/CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,15 @@ All notable changes to the Berkeley Function Calling Leaderboard will be documen
   - `microsoft/Phi-3-mini-128k-instruct`
   - `microsoft/Phi-3-mini-4k-instruct`
 - [Sept 25, 2024] [#660](https://github.com/ShishirPatil/gorilla/pull/660): Bug fix in `parse_nested_value` function to handle nested dictionary values properly. 
+- [Sept 24, 2024] [#648](https://github.com/ShishirPatil/gorilla/pull/648): Add the following new models to the leaderboard:
+  - `gemini-1.5-pro-002`
+  - `gemini-1.5-pro-002-FC`
+  - `gemini-1.5-pro-001`
+  - `gemini-1.5-pro-001-FC`
+  - `gemini-1.5-flash-002`
```
Comment (naseer9shaik) [inline berkeley-function-call-leaderboard/CHANGELOG.md:19]: Please add: 
 - `gemini-1.5-flash-001`
  - `gemini-1.5-flash-001-FC`

**Review by naseer9shaik — APPROVED** [review]

**Review by naseer9shaik — APPROVED** [review]

**Review by ShishirPatil — APPROVED** [review]

**Pull request merged**